pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -0,0 +1,188 @@
+
+	log.Info("Create or update subnetport", "subnetport", obj)
+
+	// TODO: For now, only the parent subnet is supported.
+	specSubnet := obj.Spec.Subnet
+
+	nsxSubnetPath, err := service.GetPathFromSubnet(specSubnet, obj.Namespace, service.Client)
+	if err != nil {
+		log.Error(err, "failed to get NSX resource path from subnet", "subnet", specSubnet, "namespace", obj.Namespace)
+		return err
+	}
+	obj.Status.NSXResourcePath = fmt.Sprintf("%s/ports/%s", nsxSubnetPath, obj.UID)
+
+	nsxSubnetPort, err := service.buildSubnetPort(obj)
+	if err != nil {
+		log.Error(err, "failed to build NSX subnet port")
+		return err
+	}
+
+	paras := strings.Split(nsxSubnetPath, "/")
+	nsxOrgID := paras[2]
+	nsxProjectID := paras[4]
+	nsxVPCID := paras[6]
+	nsxSubnetID := paras[8]
+
+	existingSubnetPort := service.SubnetPortStore.GetByKey(*nsxSubnetPort.Id)
+	isChanged := common.CompareResource(SubnetPortToComparable(existingSubnetPort), SubnetPortToComparable(nsxSubnetPort))
+	if !isChanged {
+		log.Info("NSX subnet port not changed, skipping the update", "nsxSubnetPort.Id", nsxSubnetPort.Id)
+		return nil
+	}
+	err = service.NSXClient.PortClient.Patch(nsxOrgID, nsxProjectID, nsxVPCID, nsxSubnetID, *nsxSubnetPort.Id, *nsxSubnetPort)
+	if err != nil {
+		log.Error(err, "failed to create or update subnet port")
+		return err
+	}
+	err = service.SubnetPortStore.Operate(nsxSubnetPort)
+	if err != nil {
+		return err
+	}
+	log.Info("successfully updated NSX subnet port", "subnetPort", nsxSubnetPort)
+	nsxSubnetPortPath := fmt.Sprintf("%s/ports/%s", nsxSubnetPath, obj.UID)
+	obj.Status.NSXResourcePath = nsxSubnetPortPath
+	return nil
+}
+
+func (service *SubnetPortService) GetPathFromSubnet(subnetName string, subnetNamespace string, client client.Client) (string, error) {
+	path := ""
+	subnet := &v1alpha1.Subnet{}
+	namespacedName := types.NamespacedName{
+		Name:      subnetName,
+		Namespace: subnetNamespace,
+	}
+	if err := client.Get(context.Background(), namespacedName, subnet); err != nil {
+		log.Error(err, "subnet CR not found", "subnet CR", namespacedName)
+		return path, err
+	}
+	path = subnet.Status.NSXResourcePath
+	if len(path) == 0 {
+		err := errors.New("empty NSX resource path from subnet")
+		return path, err
+	}
+	return path, nil
+}
+
+func (service *SubnetPortService) GetSubnetPortState(obj *v1alpha1.SubnetPort) (*model.SegmentPortState, error) {
+
+	// TODO: For now, only the parent subnet is supported.
+	specSubnet := obj.Spec.Subnet
+
+	// TODO: Get the subnet's NSX resource path from subnet store after the store is implemented.
+	nsxSubnetPath, err := service.GetPathFromSubnet(specSubnet, obj.Namespace, service.Client)
+	if err != nil {
+		log.Error(err, "failed to get NSX resource path from subnet", "subnet", specSubnet, "namespace", obj.Namespace)
+	}
+
+	paras := strings.Split(nsxSubnetPath, "/")
+	nsxOrgID := paras[2]
+	nsxProjectID := paras[4]
+	nsxVPCID := paras[6]
+	nsxSubnetID := paras[8]
```

Comment:
Or make another util function to get these paths

Reply:
done